ports/nrf/common-hal/pulseio/PulseOut.c

```diff
@@ -74,6 +74,11 @@ static void pulseout_event_handler(nrf_timer_event_t event_type, void *p_context
     nrfx_timer_pause(timer);
 
     pulse_array_index++;
+    // Ignore a zero-length pulse
+    while (pulse_array[pulse_array_index] == 0 &&
+           pulse_array_index < pulse_array_length) {
+        pulse_array_index++;
+    }
```

Comment:
shouldn't the two halves of this test be in the other order, so that the length is checked first?  otherwise, there's the possibility of reading one item past the end of the pulse array.

Reply:
Yes, I see your point. I generally write the less likely expression first to speed things up, but that doesn't work here. Change has been made as suggested.